modules/basicfuncs/tests/test_basicfuncs.c

```diff
@@ -136,6 +146,23 @@ Test(basicfuncs, test_str_funcs)
 {
   assert_template_format("$(ipv4-to-int $SOURCEIP)", "168496141");
 
+  if (_is_connected_to_the_internet())
+    {
+      /* Note: The dns-resolve-ip tests rely on each other, by setting/using the dns-cache */
+      assert_template_format("$(dns-resolve-ip --use-dns=no --dns-cache=no 198.41.0.4)", "198.41.0.4");
```

Comment:
Regarding the tests I would consider the followings:
* stub the actual libc call resolving ip to dns (test should not rely on internet, or network in fact. even if you check if there is internet you could still get failed case due to faulty network, and God help us investigating those ... )
* Split at least dependent tests/assert into their own (as from your comment it is far from clear which tests are dependent, or all of them ?!)

Reply:
I agree! Might be possible to do the same with `lib/tests/test_host_resolve.c`, which (per my understanding) does not run with `make check`, because of the network dependency.
I will move the dependent TCs as well.

Reply:
I stubbed on a higher layer, I think it is okay to do so.
Also, now as `use-dns=no` and `use-dns-cache=yes` together are not valid options, we do not have dependent TCs.